core/src/main/java/org/bitcoinj/core/TransactionConfidence.java

```diff
@@ -337,23 +339,37 @@ public boolean wasBroadcastBy(PeerAddress address) {
     }
 
     /** Return the time the transaction was last announced to us. */
-    public Date getLastBroadcastedAt() {
+    @Nullable
+    public Instant getLastBroadcastedAtInstant() {
         return lastBroadcastedAt;
     }
 
+    /** @deprecated use {@link #getLastBroadcastedAtInstant()} */
+    @Deprecated
+    @Nullable
+    public Date getLastBroadcastedAt() {
+        return lastBroadcastedAt != null ? Date.from(lastBroadcastedAt) : null;
+    }
+
     /** Set the time the transaction was last announced to us. */
-    public void setLastBroadcastedAt(Date lastBroadcastedAt) {
+    public void setLastBroadcastedAt(Instant lastBroadcastedAt) {
```

Comment:
Maybe name this `setBroadcastTime`?  Also annotate as `@Nullable`?

Actually, could we not allow it to be set to `null`? Maybe it's only initialized to `null` and the only valid use case is "set to an actual time"?

Reply:
Its not really meant to use null (clear the time). I'll add a `checkNotNull()`. I'll also rename to your suggestion.